### ISSUE TITLE
RF01 false positive for Azure Synapse SQL OPENROWSET tables

### ISSUE THREAD

**Issue by daviewales** [sqlfluff/sqlfluff#6778]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`sqlfluff lint` raises an `RF01` lint error when linting a select statement loaded from an Azure Synapse `OPENROWSET` function.

### Expected Behaviour

No lint error should be raised.

### Observed Behaviour

```
$ sqlfluff lint --ignore-local-config -d tsql test.sql
== [test.sql] FAIL
L:   1 | P:   8 | RF01 | Reference 'result.test_field' refers to table/view not
                       | found in the FROM clause or found in ancestor statement.
                       | [references.from]
```

### How to reproduce

``` sql
-- test.sql
SELECT result.test_field
FROM
    OPENROWSET (
        BULK 'https://example.dfs.core.windows.net/container/test.parquet',
        FORMAT = 'PARQUET'
    ) AS [result]
```

Command to reproduce:
``` sh
sqlfluff lint --ignore-local-config -d tsql test.sql
```

### Dialect

tsql

### Version

```
sqlfluff, version 3.3.1
Python 3.12.5
```

### Configuration

Default config (`--ignore-local-config -d tsql`)

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
This has been fixed in #6702 and will be part of the next release
